Dart/src/com/jetbrains/lang/dart/ide/devtools/DevToolsService.kt

```diff
@@ -0,0 +1,87 @@
+// Copyright 2000-2024 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.jetbrains.lang.dart.ide.devtools
+
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.ColoredProcessHandler
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessListener
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.openapi.util.text.StringUtil
+import com.jetbrains.lang.dart.sdk.DartSdk
+import io.flutter.run.daemon.DevToolsInstance
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.CompletableFuture
+
+@Service(Service.Level.PROJECT)
+class DevToolsService(private val myProject: Project) : Disposable {
+  private val MIN_SDK_VERSION: String = "2.15"
+  private lateinit var processHandler: ColoredProcessHandler;
+
+  private val devToolsFuture: CompletableFuture<DevToolsInstance> = CompletableFuture()
+
+  fun startService(dtdUri: String? = null): Void? {
+    val sdk = DartSdk.getDartSdk(myProject)
+    if (sdk == null || !isDartSdkVersionSufficient(sdk)) {
+      return null
+    }
+    val result = GeneralCommandLine().withWorkDirectory(sdk.homePath)
+    result.charset = StandardCharsets.UTF_8
+    result.exePath = FileUtil.toSystemDependentName("dart")
+    result.addParameter("devtools")
+    result.addParameter("--machine")
+    if (dtdUri != null) {
+      result.addParameter("--dtd-uri=$dtdUri")
```

Comment:
I'm getting an error `Could not find an option named "dtd-uri"`
Dart SDK 3.4.0-194.0.dev built 5 Mar 2024

Was this option introduced later?

Reply:
This should be included, e.g. I have `Dart SDK version: 3.4.0-171.0.dev` and can run `dart devtools --machine --dtd-uri=something`. Are you also getting this error when you run the command on your command line?

Reply:
Got it.
The code that starts DevTools is
```
val result = GeneralCommandLine().withWorkDirectory(sdk.homePath)
result.exePath = FileUtil.toSystemDependentName("dart")
```
`sdk` is correct for me, it is 3.4.0-194.0.dev. But the exePath is just "dart", so `dart` executable is taken from `$PATH` (having `withWorkDirectory(sdk.homePath)` doesn't help). For me it is `/opt/homebrew/bin/dart` which is some old SDK version.

I fixed it by using 
```
commandLine.exePath = FileUtil.toSystemDependentName(DartSdkUtil.getDartExePath(sdk))
```
like in DartToolingDaemonService.


BTW, does process home directory matter (for DTD or DT)?